Rebase to 2.41.0 rc1

mjcheetham opened the pull request:
Range-diff relative to `main`:

```
  1:  ff2d9865eb =   1:  ef0abed3ee windows: ignore empty `PATH` elements
  2:  3e26f8b1b2 =   2:  e73c2927c4 is_Cygwin: avoid `exec`ing anything
  3:  b846021c8f =   3:  d6e50bb3e4 Move is_<platform> functions to the beginning
  4:  aa3d45225e =   4:  f133efb718 Move the `_which` function (almost) to the top
  5:  c9990db802 =   5:  15492bc3be Work around Tcl's default `PATH` lookup
  6:  8209918221 =   6:  88d856154a gitk(Windows): avoid inadvertently calling executables in the worktree
  7:  964e74d506 =   7:  b38a10cb5b t9350: point out that refs are not updated correctly
  8:  e3087b0f5f =   8:  fd5b3d228b transport-helper: add trailing --
 13:  250dbf3626 =   9:  20b9632685 mingw: include the Python parts in the build
  9:  ae687db89c =  10:  ba98482e49 remote-helper: check helper status after import/export
 10:  661307cea2 =  11:  e1db8f2e53 mingw: demonstrate a problem with certain absolute paths
 16:  1d0d360357 =  12:  8facb32f75 clean: do not traverse mount points
 14:  d54659d3aa =  13:  cf615e8797 win32/pthread: avoid name clashes with winpthread
 11:  989acfa0b6 =  14:  bd83d13920 Always auto-gc after calling a fast-import transport
 12:  7cc62a6e3e =  15:  22cccd5ba5 mingw: allow absolute paths without drive prefix
 18:  6a039c6cc3 =  16:  e0e0fb321e clean: remove mount points when possible
 15:  2cc9779ac9 =  17:  087d939e36 git-compat-util: avoid redeclaring _DEFAULT_SOURCE
 17:  248cf0faf0 =  18:  480ae6dd36 Import the source code of mimalloc v2.0.9
 19:  ab5288681a =  19:  031d1ae6e7 mimalloc: adjust for building inside Git
 20:  792aee5aa7 =  20:  d3a98a35d8 mimalloc: offer a build-time option to enable it
114:  f44388cf4e =  21:  fea16f542d mimalloc: use "weak" random seed when statically linked
115:  80e7344829 =  22:  30c73c45cc mingw: use mimalloc
 21:  0e850676b8 =  23:  cb9d01a120 Config option to disable side-band-64k for transport
 22:  e499919ab3 =  24:  96c09c3384 mingw: make sure `errno` is set correctly when socket operations fail
 23:  68a71697ad =  25:  6b51f8e293 mingw: do resolve symlinks in `getcwd()`
 24:  de660602c8 =  26:  bbbcbe409a mingw: fix fatal error working on mapped network drives on Windows
 33:  3b3d99b50a =  27:  a1eedfa0d0 vcxproj: unclash project directories with build outputs
 25:  5334e5bb75 =  28:  e94869ffe5 mingw: ensure valid CTYPE
 26:  ac86f23e23 =  29:  902940aa65 mingw: demonstrate a `git add` issue with NTFS junctions
 34:  2c2ece9971 =  30:  384691edf4 t5505/t5516: allow running without `.git/branches/` in the templates
 27:  36060c9fb7 =  31:  7d6943e6a6 mingw: allow `git.exe` to be used instead of the "Git wrapper"
 28:  e103d85454 =  32:  fc47bbc7de strbuf_realpath(): use platform-dependent API if available
 35:  1bfabd2cb3 =  33:  3548998449 t5505/t5516: fix white-space around redirectors
 29:  6b222d47dc =  34:  0de8a64712 mingw: ignore HOMEDRIVE/HOMEPATH if it points to Windows' system directory
 30:  40b9818b04 =  35:  af0579a6f9 http: use new "best effort" strategy for Secure Channel revoke checking
 31:  eefc355e11 =  36:  de14f9752f clink.pl: fix MSVC compile script to handle libcurl-d.lib
 32:  dd06757e3e =  37:  78779fd231 mingw: implement a platform-specific `strbuf_realpath()`
 39:  7652aa7f6d =  38:  279774ee6b clink.pl: fix libexpatd.lib link error when using MSVC
 40:  a9114bfdd6 =  39:  4dd2270bc0 Makefile: clean up .ilk files when MSVC=1
 41:  05c81bc3ca =  40:  92a7786dd0 vcbuild: add support for compiling Windows resource files
 36:  116cfb8cff =  41:  a1700b1d7e t3701: verify that we can add *lots* of files interactively
 37:  edcf3d2ed1 =  42:  ad8fe6f9e3 git add -i: handle CR/LF line endings in the interactive input
 38:  1835eac1de =  43:  16ff19b753 commit: accept "scissors" with CR/LF line endings
 42:  526d1589df =  44:  6fed173716 t0014: fix indentation
 43:  913c2dc61c =  45:  93a7ca3fd2 git-gui: accommodate for intent-to-add files
 44:  a744c63eb6 =  46:  17839eeeb0 config.mak.uname: add git.rc to MSVC builds
 45:  ceed8218d7 =  47:  0c1018855a clink.pl: ignore no-stack-protector arg on MSVC=1 builds
 46:  ede97ba731 =  48:  dec0be4d72 clink.pl: move default linker options for MSVC=1 builds
 47:  b3f46dff67 =  49:  731f0eeaa1 buildsystems: remove duplicate clause
 48:  ab08446e41 =  50:  1ce0f27c15 vcxproj: handle resource files, too
 49:  81a1cb8707 =  51:  1879d42add vcxproj: ignore -fno-stack-protector and -fno-common
 50:  4bab165263 =  52:  893d80ce56 vcxproj: handle GUI programs, too
 51:  8fc6dd1955 =  53:  fd69bfd057 win32: add a helper to run `git.exe` without a foreground window
 52:  258d3e556b =  54:  744e2c28b8 git maintenance: avoid console window in scheduled tasks on Windows
 53:  010623dab7 =  55:  a5e48cf73b vcpkg_install: detect lack of Git
 60:  cad90d9721 =  56:  1d086b2b23 cmake: install headless-git.
 54:  84ffa03433 =  57:  4cce9186dd vcpkg_install: add comment regarding slow network connections
 55:  60e1a76108 =  58:  f63b9076a8 vcxproj: support building Windows/ARM64 binaries
 56:  21350c7250 =  59:  4a3c1b03ff vcbuild: install ARM64 dependencies when building ARM64 binaries
 57:  14a0d892da =  60:  b9e46809ec vcbuild: add an option to install individual 'features'
 58:  abc0841007 =  61:  528ad1b927 cmake: allow building for Windows/ARM64
 59:  7583106a68 =  62:  6103b685e6 ci(vs-build) also build Windows/ARM64 artifacts
 61:  373cedd527 =  63:  d50ba7343e Add schannel to curl installation
 63:  cbee15fe81 =  64:  a489baa002 cmake(): allow setting HOST_CPU for cross-compilation
 64:  3305479699 =  65:  80051ff783 ci(vs-build): download the vcpkg artifacts using a dedicated Action
 65:  2b88463839 =  66:  f6d455806e mingw: allow for longer paths in `parse_interpreter()`
 66:  aa8dc1e345 =  67:  d216b539aa compat/vcbuild: document preferred way to build in Visual Studio
 69:  ed6e2fd6ef =  68:  a3ec537de4 CMake: default Visual Studio generator has changed
 70:  9e3c24b588 =  69:  b5b4b838ad hash-object: demonstrate a >4GB/LLP64 problem
 71:  02db910682 =  70:  d18dfa1a75 .gitignore: add Visual Studio CMakeSetting.json file
 72:  dd6f636a71 =  71:  7680076547 hash_object_file_literally(): use size_t
 62:  099e5a4b05 =  72:  672e522ba0 subtree: update `contrib/subtree` `test` target
 73:  345dd6baba =  73:  48c4b51b39 CMakeLists: add default "x64-windows" arch for Visual Studio
 74:  3455e87eb6 =  74:  9f067d59d7 object-file.c: use size_t for header lengths
 67:  bbf1d39cbb =  75:  a14ca975e6 http: optionally send SSL client certificate
 68:  adc7f2015d =  76:  0fc293e71c ci: run `contrib/subtree` tests in CI builds
 75:  a80401f59c =  77:  731c499894 CMake: show Win32 and Generator_platform build-option values
 76:  5148cfc0bc =  78:  462855f05f init: do parse _all_ core.* settings early
 77:  906adbad15 =  79:  0c38330b28 Enable the built-in FSMonitor as an experimental feature
 78:  305c7e3616 =  80:  603456cbd4 hash algorithms: use size_t for section lengths
 79:  3288c0dd3a =  81:  0b3a0e1134 hash-object --stdin: verify that it works with >4GB/LLP64
 80:  17ebeea7b2 =  82:  161a829cd1 hash-object: add another >4GB/LLP64 test case
 81:  f330be2610 =  83:  fa03e4440a setup: properly use "%(prefix)/" when in WSL
 88:  162632fb0d =  84:  9438edeacf Add config option `windows.appendAtomically`
 82:  846c3061ff =  85:  8f5e4c2eb5 hash-object: add a >4GB/LLP64 test case using filtered input
 83:  8a7ca31f5c =  86:  d641d23208 compat/mingw.c: do not warn when failing to get owner
 84:  5d37758bcb =  87:  6cdc68ae7d mingw: $env:TERM="xterm-256color" for newer OSes
 89:  c2b9e39faa =  88:  a0fafb506c winansi: check result and Buffer before using Name
116:  3683fc052f =  89:  ec4eb13945 mingw: change core.fsyncObjectFiles = 1 by default
 85:  7c609cb499 =  90:  cab6aeef03 vcxproj: allow building with `NO_PERL` again
 86:  8029c2264a =  91:  f70de51f2b vcxproj: require C11
 87:  2994f0852e =  92:  77caf99a3e vcxproj: ignore the `-pedantic` option
 90:  ef509aebe6 =  93:  166470aae0 vcxproj: include reftable when committing `.vcxproj` files
 91:  fc3b740a79 =  94:  afbc0f7053 vcxproj: handle libreftable_test, too
 92:  5580331dbd =  95:  4d6d3a7e6a vcxproj: avoid escaping double quotes in the defines
 93:  d54a051a89 =  96:  ca4260698c ci: adjust Azure Pipeline for `runs_on_pool`
 94:  b81d5851fe =  97:  9944a9df8e ci: stop linking the `prove` cache
 95:  cf0af1cece =  98:  03be20c731 ci: reinstate Azure Pipelines support
 96:  dea6b4bc5b =  99:  5e8a990c89 azure-pipeline: drop the `GETTEXT_POISON` job
 97:  c06bcf3cc2 = 100:  46fca9814b azure-pipeline: stop hard-coding `apt-get` calls
 98:  c7685155bd = 101:  1a36960c65 azure-pipeline: drop the code to write to/read from a file share
 99:  b1b9e6f064 = 102:  88e383a82d azure-pipeline: use partial clone/parallel checkout to initialize minimal-sdk
101:  0456155be4 = 103:  bdf3800f6d azure-pipeline: downcase the job name of the `Linux32` job
104:  b0c0796453 = 104:  37a6c7fe53 azure-pipeline: run static-analysis on jammy
100:  11c9910f49 = 105:  624cd0d23f bswap.h: add support for built-in bswap functions
102:  1b038fb6cf = 106:  744310dc4f config.mak.uname: add support for clangarm64
103:  fedf6b474b = 107:  0d2efe823b MinGW: link as terminal server aware
105:  86e57b790a = 108:  252bb35b9f Fix Windows version resources
106:  0e6bc0cb32 = 109:  0852660d24 ci: create clangarm64-build.yml
107:  e391158953 = 110:  42fb4538ea status: fix for old-style submodules with commondir
110:  79ff1787c4 = 111:  363e32fcaa http: optionally load libcurl lazily
111:  a62185fe34 = 112:  72f55195cc http: support lazy-loading libcurl also on Windows
112:  380082bbb1 = 113:  31568109b8 http: when loading libcurl lazily, allow for multiple SSL backends
108:  4137e08394 = 114:  e735c95e51 windows: skip linking `git-<command>` for built-ins
109:  86fafaaffc = 115:  863aa24acb windows: fix Repository>Explore Working Copy
113:  681925b76a = 116:  93e21f029f mingw: do load libcurl dynamically by default
117:  7fba6c5230 = 117:  3c74e5d7cb Win32: make FILETIME conversion functions public
118:  cb6bc1bda4 = 118:  74c136a6f3 Win32: dirent.c: Move opendir down
119:  da945a4f6f = 119:  5932027b1a mingw: make the dirent implementation pluggable
120:  7485ed23cc = 120:  69ea3ebadd Win32: make the lstat implementation pluggable
121:  1dba7a716e = 121:  efc74c18b2 add infrastructure for read-only file system level caches
122:  496906df6f = 122:  b7ea0f9442 mingw: add a cache below mingw's lstat and dirent implementations
123:  7820919795 = 123:  2421664479 fscache: load directories only once
124:  dd1253aba6 = 124:  a9b42d3dba fscache: add key for GIT_TRACE_FSCACHE
125:  901c18a16d = 125:  abd01705b6 fscache: remember not-found directories
126:  590aa32a33 = 126:  1a1198e365 fscache: add a test for the dir-not-found optimization
127:  f0eabf5a56 = 127:  86d807797e add: use preload-index and fscache for performance
128:  29722694ed = 128:  42c399ed10 dir.c: make add_excludes aware of fscache during status
129:  ec276aa261 = 129:  27582d8832 fscache: make fscache_enabled() public
130:  794703277c = 130:  c30ed6fba5 dir.c: regression fix for add_excludes with fscache
131:  4e8c179b27 = 131:  d973451988 fetch-pack.c: enable fscache for stats under .git/objects
132:  6b196a94a5 = 132:  5e4d66bcce checkout.c: enable fscache for checkout again
133:  5593eafbe0 = 133:  f9c8d9cc8e Enable the filesystem cache (fscache) in refresh_index().
134:  dd2962c451 = 134:  b4e0e7a623 fscache: use FindFirstFileExW to avoid retrieving the short name
135:  47d4bd432f = 135:  2d8d2402f0 status: disable and free fscache at the end of the status command
136:  366a6de477 = 136:  e4d5afb7db fscache: add GIT_TEST_FSCACHE support
137:  b7f2b0e64b = 137:  b34e0ee43c fscache: add fscache hit statistics
138:  cc88edcc82 = 138:  a0364bd142 mem_pool: add GIT_TRACE_MEMPOOL support
139:  cbdde2f861 = 139:  f6595f34c1 fscache: fscache takes an initial size
140:  b09113a55a = 140:  a5f8b37468 fscache: update fscache to be thread specific instead of global
141:  57e6ec2394 = 141:  2c5320cd78 fscache: teach fscache to use mempool
142:  beb625b5fc = 142:  ec6ce9dbde fscache: make fscache_enable() thread safe
143:  d45a51a89e = 143:  cfd1116781 fscache: teach fscache to use NtQueryDirectoryFile
144:  29d7b93b94 = 144:  96ff33b44d unpack-trees: enable fscache for sparse-checkout
150:  6c7fcb2724 = 145:  de0fba5ff3 fscache: remember the reparse tag for each entry
145:  5b256c08af = 146:  c10d87882f git-gui: provide question helper for retry fallback on Windows
154:  78fd100eab = 147:  5a0dc79428 fscache: implement an FSCache-aware is_mount_point()
146:  1c73926b78 = 148:  ad59198341 git gui: set GIT_ASKPASS=git-gui--askpass if not set yet
152:  c9b45956e7 = 149:  87223a6105 respect core.hooksPath, falling back to .git/hooks
157:  8d9a82ce12 = 150:  5735819fcf clean: make use of FSCache
147:  cba3e45249 = 151:  b7c1aa7ecd gitk: Unicode file name support
148:  5e6a102a75 = 152:  d5ae9b2730 git-gui--askyesno: fix funny text wrapping
149:  1d780d2fef = 153:  85ff443be7 gitk: Use an external icon file on Windows
151:  14a10af201 = 154:  263a9bbd4d git-gui--askyesno: allow overriding the window title
153:  fe2027b742 = 155:  90c434804d gitk: fix arrow keys in input fields with Tcl/Tk >= 8.6
155:  86352a1c86 = 156:  4a4bff144f git-gui--askyesno (mingw): use Git for Windows' icon, if available
156:  1b7283f2cd = 157:  554084465b gitk: make the "list references" default window width wider
158:  f2276d5734 = 158:  ab861d3a85 pack-objects (mingw): demonstrate a segmentation fault with large deltas
159:  a510c85f17 = 159:  dcad435295 mingw: support long paths
160:  343cdd47e4 = 160:  7700d3083f Win32: fix 'lstat("dir/")' with long paths
161:  2ab4087450 = 161:  aed3c9fe5a mingw: Support `git_terminal_prompt` with more terminals
162:  7cd00c4c3f = 162:  d3c4a4015a compat/terminal.c: only use the Windows console if bash 'read -r' fails
163:  30082ccf85 = 163:  f0457c0049 mingw (git_terminal_prompt): do fall back to CONIN$/CONOUT$ method
164:  2c9a4ba781 = 164:  9ce99659a0 Unbreak interactive GPG prompt upon signing
165:  da33dfb5ad = 165:  175bc477f5 strbuf_readlink: don't call readlink twice if hint is the exact link size
166:  7260b89af0 = 166:  efb776cd78 strbuf_readlink: support link targets that exceed PATH_MAX
167:  fa4cc119d2 = 167:  9d8290e5b3 lockfile.c: use is_dir_sep() instead of hardcoded '/' checks
168:  6b98f52b55 = 168:  1da53803ad Win32: don't call GetFileAttributes twice in mingw_lstat()
169:  00670dd9a6 = 169:  907a752628 Win32: implement stat() with symlink support
170:  48e57be018 = 170:  4180d10c53 Win32: remove separate do_lstat() function
171:  05feba82d0 = 171:  8a1e5c83a6 Win32: let mingw_lstat() error early upon problems with reparse points
180:  67f80a1e6f = 172:  772266f0d9 compat/fsmonitor/fsm-*-win32: support long paths
181:  cf790a8b9b = 173:  7d96c5d4b2 clean: suggest using `core.longPaths` if paths are too long to remove
172:  b795b6de8d = 174:  e56ba15db5 mingw: teach fscache and dirent about symlinks
173:  f72f400783 = 175:  6166971c78 Win32: lstat(): return adequate stat.st_size for symlinks
174:  f01e3ed553 = 176:  85c6e571d2 Win32: factor out retry logic
175:  681b451be7 = 177:  b460afb470 Win32: change default of 'core.symlinks' to false
176:  8292e5c549 = 178:  bb95514ffa Win32: add symlink-specific error codes
177:  7b4aee06d3 = 179:  c686ad999e Win32: mingw_unlink: support symlinks to directories
178:  8cdbbc7634 = 180:  95f1c35131 Win32: mingw_rename: support renaming symlinks
179:  16c301f6a9 = 181:  42642caa68 Win32: mingw_chdir: change to symlink-resolved directory
182:  d41dd0264d = 182:  538f7204c4 Win32: implement readlink()
183:  963f37ebbf = 183:  3b39afba1d mingw: lstat: compute correct size for symlinks
184:  5e2c593fcd = 184:  a3169483d6 Win32: implement basic symlink() functionality (file symlinks only)
185:  d9c4eb4cc7 = 185:  08a3ffe6a3 Win32: symlink: add support for symlinks to directories
186:  353fe016aa = 186:  46d8cf322d mingw: try to create symlinks without elevated permissions
187:  d93222392f = 187:  c4ccc9d3c1 mingw: emulate stat() a little more faithfully
188:  b11c0b50fe = 188:  9bed36d627 mingw: special-case index entries for symlinks with buggy size
189:  a0b6a39a90 = 189:  25b42b498c Win32: symlink: move phantom symlink creation to a separate function
190:  d92bba3a6a = 190:  0365c8f6a1 Introduce helper to create symlinks that knows about index_state
191:  6a73477347 = 191:  f831bfa1f7 mingw: allow to specify the symlink type in .gitattributes
200:  cdcfa82edb = 192:  b2a2412f28 mingw: introduce code to detect whether we're inside a Windows container
192:  aeae1110e3 = 193:  21eb48ac76 Win32: symlink: add test for `symlink` attribute
202:  6fdcbf7661 = 194:  7cf9418551 mingw: when running in a Windows container, try to rename() harder
193:  72ced14f64 = 195:  2399116c1b mingw: explicitly specify with which cmd to prefix the cmdline
204:  c6aed02f32 = 196:  abaefcd709 mingw: move the file_attr_to_st_mode() function definition
194:  6a6a0d2c0d = 197:  1b1c85344e mingw: when path_lookup() failed, try BusyBox
206:  731b4bb9f4 = 198:  d1c928e9c2 mingw: Windows Docker volumes are *not* symbolic links
195:  ae83b935ae = 199:  fbf503bef7 test-lib: avoid unnecessary Perl invocation
196:  0b961a8bde = 200:  ac59834555 test-tool: learn to act as a drop-in replacement for `iconv`
197:  b7d1435728 = 201:  587427561e tests(mingw): if `iconv` is unavailable, use `test-helper --iconv`
198:  5bacebe58b = 202:  d5825f2fa1 gitattributes: mark .png files as binary
199:  4580d538ad = 203:  66ac48c33d tests: move test PNGs into t/lib-diff/
201:  77bafec88a = 204:  811af96d8f tests: only override sort & find if there are usable ones in /usr/bin/
203:  0b0770c9f6 = 205:  352e830ae5 tests: use the correct path separator with BusyBox
205:  9b4b63529e = 206:  d78071fdd0 mingw: only use Bash-ism `builtin pwd -W` when available
207:  6dee2eafd5 = 207:  054abfc876 tests (mingw): remove Bash-specific pwd option
208:  3001272ada = 208:  e7a169f0ef test-lib: add BUSYBOX prerequisite
209:  44057a9329 = 209:  4d9c05fe5c t5003: use binary file from t/lib-diff/
210:  c23961b833 = 210:  f9bec2218c t5532: workaround for BusyBox on Windows
211:  68a4bdf8ba = 211:  70f0d45ee9 t5605: special-case hardlink test for BusyBox-w32
212:  275a9350dd = 212:  5da4a016b2 t5813: allow for $PWD to be a Windows path
213:  9b5c4ed4d8 = 213:  ef929d25d8 t9200: skip tests when $PWD contains a colon
214:  c78531d0f6 = 214:  24269d899e mingw: kill child processes in a gentler way
215:  af1eece682 = 215:  e1f86baa8f mingw: do not call xutftowcs_path in mingw_mktemp
216:  f534655d9c = 216:  8b7ea10844 mingw: add a Makefile target to copy test artifacts
217:  864ec33ab4 = 217:  fb99170dc4 mingw: really handle SIGINT
222:  fefa706e7a = 218:  eb049e6adb Describe Git for Windows' architecture [no ci]
223:  2de8f76c0e = 219:  e919023eeb Modify the Code of Conduct for Git for Windows
224:  4fdce33acb = 220:  c156847b8f CONTRIBUTING.md: add guide for first-time contributors
225:  93b0207e5c = 221:  2ea97cfdf5 README.md: Add a Windows-specific preamble
218:  bd50ab65d3 = 222:  d5e229528b Partially un-revert "editor: save and reset terminal after calling EDITOR"
219:  9209cb6686 = 223:  ec3729989b reset: reinstate support for the deprecated --stdin option
220:  5ca864827f = 224:  888e5d0413 fsmonitor: reintroduce core.useBuiltinFSMonitor
221:  c58f0000c2 = 225:  9076985016 Add a GitHub workflow to monitor component updates
226:  8fc5573df1 = 226:  a4d1b6d2a4 Add an issue template
227:  00ebf09456 = 227:  efd05942dc Modify the GitHub Pull Request template (to reflect Git for Windows)
228:  9e60d9f349 = 228:  f6b45188bf .github: Add configuration for the Sentiment Bot
229:  b00c335f94 = 229:  896fa8e176 Document how $HOME is set on Windows
230:  4ac68b6a54 = 230:  05b5296bc4 SECURITY.md: document Git for Windows' policies
```

vdye reviewed (approved):
Nothing like a clean rebase! :shipit:

derrickstolee reviewed (approved):
I thought we had some `fixup!` commits that got in after the `-rc0` rebase, but they must have been _before_ that rebase. Thanks!

mjcheetham commented:
Windows build fails to due the removal of an argument from `git_check_attr` in `mingw.c` in .
The only call-site passed `NULL` for the removed arg; updated to drop it.

```
191:  6a73477347 ! 194:  871817c12a mingw: allow to specify the symlink type in .gitattributes
    @@ compat/mingw.c: int link(const char *oldpath, const char *newpath)
     +	if (!check)
     +		check = attr_check_initl("symlink", NULL);
     +
    -+	git_check_attr(index, NULL, link, check);
    ++	git_check_attr(index, link, check);
     +
     +	value = check->items[0].value;
     +	if (ATTR_UNSET(value))
```

Updated range-diff:
```
  1:  ff2d9865eb =   1:  11e5c1a6c9 windows: ignore empty `PATH` elements
  2:  3e26f8b1b2 =   2:  652d0ae241 is_Cygwin: avoid `exec`ing anything
  3:  b846021c8f =   3:  38b32f1df8 Move is_<platform> functions to the beginning
  4:  aa3d45225e =   4:  56c271cd88 Move the `_which` function (almost) to the top
  5:  c9990db802 =   5:  c45feb711b Work around Tcl's default `PATH` lookup
  6:  8209918221 =   6:  2ef7ac49b5 gitk(Windows): avoid inadvertently calling executables in the worktree
  7:  964e74d506 =   7:  1b98f33b33 t9350: point out that refs are not updated correctly
  8:  e3087b0f5f =   8:  3f83b2598e transport-helper: add trailing --
  9:  ae687db89c =   9:  88d859a346 remote-helper: check helper status after import/export
 10:  661307cea2 =  10:  3cf6cd1aaf mingw: demonstrate a problem with certain absolute paths
 16:  1d0d360357 =  11:  8e17eb72d6 clean: do not traverse mount points
 11:  989acfa0b6 =  12:  102981f5eb Always auto-gc after calling a fast-import transport
 12:  7cc62a6e3e =  13:  352fb74fd6 mingw: allow absolute paths without drive prefix
 13:  250dbf3626 =  14:  04c05d01d9 mingw: include the Python parts in the build
 14:  d54659d3aa =  15:  2b2f644650 win32/pthread: avoid name clashes with winpthread
 15:  2cc9779ac9 =  16:  b3431d4ea4 git-compat-util: avoid redeclaring _DEFAULT_SOURCE
 17:  248cf0faf0 =  17:  dcc9e4cc44 Import the source code of mimalloc v2.0.9
 18:  6a039c6cc3 =  18:  4fc0747f48 clean: remove mount points when possible
 19:  ab5288681a =  19:  0476f6ca96 mimalloc: adjust for building inside Git
 20:  792aee5aa7 =  20:  ae5271f4a7 mimalloc: offer a build-time option to enable it
114:  f44388cf4e =  21:  1ab3b65c41 mimalloc: use "weak" random seed when statically linked
115:  80e7344829 =  22:  29ab7978b8 mingw: use mimalloc
 21:  0e850676b8 =  23:  177eb9afce Config option to disable side-band-64k for transport
 22:  e499919ab3 =  24:  1a53f28a98 mingw: make sure `errno` is set correctly when socket operations fail
 23:  68a71697ad =  25:  3a979281a3 mingw: do resolve symlinks in `getcwd()`
 24:  de660602c8 =  26:  81dbbcebd9 mingw: fix fatal error working on mapped network drives on Windows
 26:  ac86f23e23 =  27:  7363ffdaa7 mingw: demonstrate a `git add` issue with NTFS junctions
 28:  e103d85454 =  28:  e9c76176d9 strbuf_realpath(): use platform-dependent API if available
 31:  eefc355e11 =  29:  7fc859f22d clink.pl: fix MSVC compile script to handle libcurl-d.lib
 32:  dd06757e3e =  30:  03e5575614 mingw: implement a platform-specific `strbuf_realpath()`
 25:  5334e5bb75 =  31:  be3afaf335 mingw: ensure valid CTYPE
 33:  3b3d99b50a =  32:  dff373849a vcxproj: unclash project directories with build outputs
 27:  36060c9fb7 =  33:  7f611a09f1 mingw: allow `git.exe` to be used instead of the "Git wrapper"
 34:  2c2ece9971 =  34:  b168cddfd9 t5505/t5516: allow running without `.git/branches/` in the templates
 29:  6b222d47dc =  35:  ee8c019053 mingw: ignore HOMEDRIVE/HOMEPATH if it points to Windows' system directory
 30:  40b9818b04 =  36:  ee1cefb837 http: use new "best effort" strategy for Secure Channel revoke checking
 35:  1bfabd2cb3 =  37:  a54dde58e9 t5505/t5516: fix white-space around redirectors
 36:  116cfb8cff =  38:  667eee2be0 t3701: verify that we can add *lots* of files interactively
 37:  edcf3d2ed1 =  39:  8daeae1ac8 git add -i: handle CR/LF line endings in the interactive input
 38:  1835eac1de =  40:  8a8473ae94 commit: accept "scissors" with CR/LF line endings
 42:  526d1589df =  41:  46815a99ae t0014: fix indentation
 43:  913c2dc61c =  42:  e099d3bbf8 git-gui: accommodate for intent-to-add files
 39:  7652aa7f6d =  43:  85157f8106 clink.pl: fix libexpatd.lib link error when using MSVC
 40:  a9114bfdd6 =  44:  d303f1ebce Makefile: clean up .ilk files when MSVC=1
 41:  05c81bc3ca =  45:  96596caead vcbuild: add support for compiling Windows resource files
 44:  a744c63eb6 =  46:  dffea57b9c config.mak.uname: add git.rc to MSVC builds
 45:  ceed8218d7 =  47:  349962ae37 clink.pl: ignore no-stack-protector arg on MSVC=1 builds
 46:  ede97ba731 =  48:  42ea893cb4 clink.pl: move default linker options for MSVC=1 builds
 47:  b3f46dff67 =  49:  52e0880890 buildsystems: remove duplicate clause
 48:  ab08446e41 =  50:  221f836d8e vcxproj: handle resource files, too
 49:  81a1cb8707 =  51:  d0b07b82e5 vcxproj: ignore -fno-stack-protector and -fno-common
 50:  4bab165263 =  52:  40b7276568 vcxproj: handle GUI programs, too
 51:  8fc6dd1955 =  53:  4507373f91 win32: add a helper to run `git.exe` without a foreground window
 52:  258d3e556b =  54:  18a68ea614 git maintenance: avoid console window in scheduled tasks on Windows
 60:  cad90d9721 =  55:  e0506bc01d cmake: install headless-git.
 53:  010623dab7 =  56:  9d2d6dd86b vcpkg_install: detect lack of Git
 54:  84ffa03433 =  57:  56acc624c1 vcpkg_install: add comment regarding slow network connections
 55:  60e1a76108 =  58:  f66e1d4f6e vcxproj: support building Windows/ARM64 binaries
 56:  21350c7250 =  59:  f6b6f5c847 vcbuild: install ARM64 dependencies when building ARM64 binaries
 57:  14a0d892da =  60:  d2bdbe7620 vcbuild: add an option to install individual 'features'
 58:  abc0841007 =  61:  8dda1d2763 cmake: allow building for Windows/ARM64
 59:  7583106a68 =  62:  214e3cd210 ci(vs-build) also build Windows/ARM64 artifacts
 61:  373cedd527 =  63:  c09d14299a Add schannel to curl installation
 63:  cbee15fe81 =  64:  f16ec5e458 cmake(): allow setting HOST_CPU for cross-compilation
 64:  3305479699 =  65:  c8df8c0c73 ci(vs-build): download the vcpkg artifacts using a dedicated Action
 65:  2b88463839 =  66:  50d551d818 mingw: allow for longer paths in `parse_interpreter()`
 66:  aa8dc1e345 =  67:  12dc57acd2 compat/vcbuild: document preferred way to build in Visual Studio
 67:  bbf1d39cbb =  68:  d0a487bf8d http: optionally send SSL client certificate
 69:  ed6e2fd6ef =  69:  0394077742 CMake: default Visual Studio generator has changed
 71:  02db910682 =  70:  a4d218f941 .gitignore: add Visual Studio CMakeSetting.json file
 62:  099e5a4b05 =  71:  bd6496131c subtree: update `contrib/subtree` `test` target
 73:  345dd6baba =  72:  b5e4daf6a2 CMakeLists: add default "x64-windows" arch for Visual Studio
 68:  adc7f2015d =  73:  e18c0ab3a0 ci: run `contrib/subtree` tests in CI builds
 75:  a80401f59c =  74:  dc0834d032 CMake: show Win32 and Generator_platform build-option values
 76:  5148cfc0bc =  75:  ca81b777fd init: do parse _all_ core.* settings early
 70:  9e3c24b588 =  76:  c04e368bcf hash-object: demonstrate a >4GB/LLP64 problem
 72:  dd6f636a71 =  77:  68831bb7c5 hash_object_file_literally(): use size_t
 74:  3455e87eb6 =  78:  ae852158a3 object-file.c: use size_t for header lengths
 78:  305c7e3616 =  79:  251a369e06 hash algorithms: use size_t for section lengths
 79:  3288c0dd3a =  80:  4b3464e717 hash-object --stdin: verify that it works with >4GB/LLP64
 80:  17ebeea7b2 =  81:  eee7353478 hash-object: add another >4GB/LLP64 test case
 77:  906adbad15 =  82:  450fdfb51a Enable the built-in FSMonitor as an experimental feature
 82:  846c3061ff =  83:  df768c7a89 hash-object: add a >4GB/LLP64 test case using filtered input
 81:  f330be2610 =  84:  facab57837 setup: properly use "%(prefix)/" when in WSL
 88:  162632fb0d =  85:  d4ffd66b8d Add config option `windows.appendAtomically`
 83:  8a7ca31f5c =  86:  06cc7d82d6 compat/mingw.c: do not warn when failing to get owner
 84:  5d37758bcb =  87:  d11f1c2665 mingw: $env:TERM="xterm-256color" for newer OSes
 89:  c2b9e39faa =  88:  6b1b983ca4 winansi: check result and Buffer before using Name
116:  3683fc052f =  89:  8a10983a62 mingw: change core.fsyncObjectFiles = 1 by default
 85:  7c609cb499 =  90:  94dce230dd vcxproj: allow building with `NO_PERL` again
 86:  8029c2264a =  91:  16bf7bc444 vcxproj: require C11
 87:  2994f0852e =  92:  93c88ccd7a vcxproj: ignore the `-pedantic` option
 90:  ef509aebe6 =  93:  b717dd3201 vcxproj: include reftable when committing `.vcxproj` files
 91:  fc3b740a79 =  94:  646d1037ea vcxproj: handle libreftable_test, too
 92:  5580331dbd =  95:  773e582c2d vcxproj: avoid escaping double quotes in the defines
 93:  d54a051a89 =  96:  672b030af4 ci: adjust Azure Pipeline for `runs_on_pool`
 94:  b81d5851fe =  97:  c045177027 ci: stop linking the `prove` cache
 95:  cf0af1cece =  98:  61c9341a37 ci: reinstate Azure Pipelines support
 96:  dea6b4bc5b =  99:  b9104c9031 azure-pipeline: drop the `GETTEXT_POISON` job
 97:  c06bcf3cc2 = 100:  590b3086f6 azure-pipeline: stop hard-coding `apt-get` calls
 98:  c7685155bd = 101:  e570e0c880 azure-pipeline: drop the code to write to/read from a file share
 99:  b1b9e6f064 = 102:  ce01e6cdb4 azure-pipeline: use partial clone/parallel checkout to initialize minimal-sdk
100:  11c9910f49 = 103:  7ce55efc3a bswap.h: add support for built-in bswap functions
101:  0456155be4 = 104:  f92b5a3f85 azure-pipeline: downcase the job name of the `Linux32` job
103:  fedf6b474b = 105:  49ec03ca47 MinGW: link as terminal server aware
102:  1b038fb6cf = 106:  9847615281 config.mak.uname: add support for clangarm64
104:  b0c0796453 = 107:  af8d06e9c9 azure-pipeline: run static-analysis on jammy
105:  86e57b790a = 108:  5cd38d6f2a Fix Windows version resources
110:  79ff1787c4 = 109:  9072a99741 http: optionally load libcurl lazily
111:  a62185fe34 = 110:  a96b932f75 http: support lazy-loading libcurl also on Windows
112:  380082bbb1 = 111:  0e6ba54459 http: when loading libcurl lazily, allow for multiple SSL backends
106:  0e6bc0cb32 = 112:  e99ab11087 ci: create clangarm64-build.yml
107:  e391158953 = 113:  74d51b4b75 status: fix for old-style submodules with commondir
108:  4137e08394 = 114:  97f50cb042 windows: skip linking `git-<command>` for built-ins
109:  86fafaaffc = 115:  a4c885096d windows: fix Repository>Explore Working Copy
113:  681925b76a = 116:  c8beedea0c mingw: do load libcurl dynamically by default
117:  7fba6c5230 = 117:  1c43aa2f80 Win32: make FILETIME conversion functions public
118:  cb6bc1bda4 = 118:  daba14e352 Win32: dirent.c: Move opendir down
119:  da945a4f6f = 119:  26ff212e5b mingw: make the dirent implementation pluggable
120:  7485ed23cc = 120:  8a45073bd9 Win32: make the lstat implementation pluggable
121:  1dba7a716e = 121:  37448c0bd3 add infrastructure for read-only file system level caches
122:  496906df6f = 122:  1bf9db1ab2 mingw: add a cache below mingw's lstat and dirent implementations
123:  7820919795 = 123:  445594b3f4 fscache: load directories only once
124:  dd1253aba6 = 124:  09f704c910 fscache: add key for GIT_TRACE_FSCACHE
125:  901c18a16d = 125:  638aa4dd7b fscache: remember not-found directories
126:  590aa32a33 = 126:  0c799e8fa6 fscache: add a test for the dir-not-found optimization
127:  f0eabf5a56 = 127:  6b8a4731ef add: use preload-index and fscache for performance
128:  29722694ed = 128:  6a6489bf50 dir.c: make add_excludes aware of fscache during status
129:  ec276aa261 = 129:  7a2be4a694 fscache: make fscache_enabled() public
130:  794703277c = 130:  f0047ace03 dir.c: regression fix for add_excludes with fscache
131:  4e8c179b27 = 131:  72f1796de8 fetch-pack.c: enable fscache for stats under .git/objects
132:  6b196a94a5 = 132:  9d8c44283a checkout.c: enable fscache for checkout again
133:  5593eafbe0 = 133:  e68a1964cc Enable the filesystem cache (fscache) in refresh_index().
134:  dd2962c451 = 134:  56db40a453 fscache: use FindFirstFileExW to avoid retrieving the short name
135:  47d4bd432f = 135:  ca67929700 status: disable and free fscache at the end of the status command
136:  366a6de477 = 136:  feb0891c00 fscache: add GIT_TEST_FSCACHE support
137:  b7f2b0e64b = 137:  f88b63fcd2 fscache: add fscache hit statistics
138:  cc88edcc82 = 138:  3f877b34cb mem_pool: add GIT_TRACE_MEMPOOL support
139:  cbdde2f861 = 139:  4a927e122f fscache: fscache takes an initial size
140:  b09113a55a = 140:  eb1cddbb53 fscache: update fscache to be thread specific instead of global
141:  57e6ec2394 = 141:  ad3b55a55c fscache: teach fscache to use mempool
142:  beb625b5fc = 142:  5089621435 fscache: make fscache_enable() thread safe
143:  d45a51a89e = 143:  3c48c4bffa fscache: teach fscache to use NtQueryDirectoryFile
144:  29d7b93b94 = 144:  b2b95aa266 unpack-trees: enable fscache for sparse-checkout
150:  6c7fcb2724 = 145:  1840da0881 fscache: remember the reparse tag for each entry
154:  78fd100eab = 146:  73988bf058 fscache: implement an FSCache-aware is_mount_point()
157:  8d9a82ce12 = 147:  098e11a2b3 clean: make use of FSCache
145:  5b256c08af = 148:  abca8483ba git-gui: provide question helper for retry fallback on Windows
146:  1c73926b78 = 149:  d713ec63b8 git gui: set GIT_ASKPASS=git-gui--askpass if not set yet
148:  5e6a102a75 = 150:  e8829e13f2 git-gui--askyesno: fix funny text wrapping
152:  c9b45956e7 = 151:  f2c683f497 respect core.hooksPath, falling back to .git/hooks
147:  cba3e45249 = 152:  28a53eff30 gitk: Unicode file name support
151:  14a10af201 = 153:  cc44581727 git-gui--askyesno: allow overriding the window title
149:  1d780d2fef = 154:  cc145a64b4 gitk: Use an external icon file on Windows
155:  86352a1c86 = 155:  0a1402ca41 git-gui--askyesno (mingw): use Git for Windows' icon, if available
153:  fe2027b742 = 156:  f9da0a3e1e gitk: fix arrow keys in input fields with Tcl/Tk >= 8.6
156:  1b7283f2cd = 157:  e2883d6b4a gitk: make the "list references" default window width wider
158:  f2276d5734 = 158:  7af2483631 pack-objects (mingw): demonstrate a segmentation fault with large deltas
159:  a510c85f17 = 159:  4b83342b54 mingw: support long paths
160:  343cdd47e4 = 160:  03b19c5c80 Win32: fix 'lstat("dir/")' with long paths
161:  2ab4087450 = 161:  99449a4283 mingw: Support `git_terminal_prompt` with more terminals
162:  7cd00c4c3f = 162:  10bdb171c8 compat/terminal.c: only use the Windows console if bash 'read -r' fails
163:  30082ccf85 = 163:  2c90056f29 mingw (git_terminal_prompt): do fall back to CONIN$/CONOUT$ method
164:  2c9a4ba781 = 164:  c4da7849db Unbreak interactive GPG prompt upon signing
180:  67f80a1e6f = 165:  22eee9a574 compat/fsmonitor/fsm-*-win32: support long paths
181:  cf790a8b9b = 166:  d0c9933de9 clean: suggest using `core.longPaths` if paths are too long to remove
165:  da33dfb5ad = 167:  a71114c721 strbuf_readlink: don't call readlink twice if hint is the exact link size
166:  7260b89af0 = 168:  52c88db3a4 strbuf_readlink: support link targets that exceed PATH_MAX
167:  fa4cc119d2 = 169:  9c5fe34020 lockfile.c: use is_dir_sep() instead of hardcoded '/' checks
168:  6b98f52b55 = 170:  a732ac9aa1 Win32: don't call GetFileAttributes twice in mingw_lstat()
169:  00670dd9a6 = 171:  b8b926a1f9 Win32: implement stat() with symlink support
170:  48e57be018 = 172:  c242a2c015 Win32: remove separate do_lstat() function
171:  05feba82d0 = 173:  e7b11cb84e Win32: let mingw_lstat() error early upon problems with reparse points
172:  b795b6de8d = 174:  f0f52c211f mingw: teach fscache and dirent about symlinks
173:  f72f400783 = 175:  130cac6e9e Win32: lstat(): return adequate stat.st_size for symlinks
174:  f01e3ed553 = 176:  abe3974227 Win32: factor out retry logic
175:  681b451be7 = 177:  dba7371ac8 Win32: change default of 'core.symlinks' to false
176:  8292e5c549 = 178:  08b3b53e08 Win32: add symlink-specific error codes
177:  7b4aee06d3 = 179:  5e67d3e859 Win32: mingw_unlink: support symlinks to directories
178:  8cdbbc7634 = 180:  8c7b9e4607 Win32: mingw_rename: support renaming symlinks
179:  16c301f6a9 = 181:  9b6ad408da Win32: mingw_chdir: change to symlink-resolved directory
182:  d41dd0264d = 182:  5410074ba5 Win32: implement readlink()
183:  963f37ebbf = 183:  57a747742d mingw: lstat: compute correct size for symlinks
184:  5e2c593fcd = 184:  fbf85ba84b Win32: implement basic symlink() functionality (file symlinks only)
185:  d9c4eb4cc7 = 185:  29365a9625 Win32: symlink: add support for symlinks to directories
186:  353fe016aa = 186:  e10e57755f mingw: try to create symlinks without elevated permissions
187:  d93222392f = 187:  f95e65a783 mingw: emulate stat() a little more faithfully
188:  b11c0b50fe = 188:  daf72c732f mingw: special-case index entries for symlinks with buggy size
200:  cdcfa82edb = 189:  b08171191e mingw: introduce code to detect whether we're inside a Windows container
202:  6fdcbf7661 = 190:  e903be5b97 mingw: when running in a Windows container, try to rename() harder
189:  a0b6a39a90 = 191:  c4ac341de9 Win32: symlink: move phantom symlink creation to a separate function
190:  d92bba3a6a = 192:  946cdd9a36 Introduce helper to create symlinks that knows about index_state
204:  c6aed02f32 = 193:  6f6bbab358 mingw: move the file_attr_to_st_mode() function definition
191:  6a73477347 ! 194:  871817c12a mingw: allow to specify the symlink type in .gitattributes
    @@ compat/mingw.c: int link(const char *oldpath, const char *newpath)
     +	if (!check)
     +		check = attr_check_initl("symlink", NULL);
     +
    -+	git_check_attr(index, NULL, link, check);
    ++	git_check_attr(index, link, check);
     +
     +	value = check->items[0].value;
     +	if (ATTR_UNSET(value))
206:  731b4bb9f4 = 195:  04e052db93 mingw: Windows Docker volumes are *not* symbolic links
192:  aeae1110e3 = 196:  0240cd3813 Win32: symlink: add test for `symlink` attribute
193:  72ced14f64 = 197:  6dcc2fc767 mingw: explicitly specify with which cmd to prefix the cmdline
194:  6a6a0d2c0d = 198:  8235698271 mingw: when path_lookup() failed, try BusyBox
195:  ae83b935ae = 199:  e67d499c28 test-lib: avoid unnecessary Perl invocation
196:  0b961a8bde = 200:  3c4dd79378 test-tool: learn to act as a drop-in replacement for `iconv`
197:  b7d1435728 = 201:  6595d38f7f tests(mingw): if `iconv` is unavailable, use `test-helper --iconv`
198:  5bacebe58b = 202:  0c8e74c05a gitattributes: mark .png files as binary
199:  4580d538ad = 203:  4acd3bc5e6 tests: move test PNGs into t/lib-diff/
201:  77bafec88a = 204:  9bee9d3ccb tests: only override sort & find if there are usable ones in /usr/bin/
203:  0b0770c9f6 = 205:  367187c3ce tests: use the correct path separator with BusyBox
205:  9b4b63529e = 206:  be7e6b241e mingw: only use Bash-ism `builtin pwd -W` when available
207:  6dee2eafd5 = 207:  d91f2b290e tests (mingw): remove Bash-specific pwd option
208:  3001272ada = 208:  65cfce033d test-lib: add BUSYBOX prerequisite
209:  44057a9329 = 209:  f988f65c41 t5003: use binary file from t/lib-diff/
210:  c23961b833 = 210:  d7ff2324fe t5532: workaround for BusyBox on Windows
211:  68a4bdf8ba = 211:  2ca9529a04 t5605: special-case hardlink test for BusyBox-w32
212:  275a9350dd = 212:  31765ca7ae t5813: allow for $PWD to be a Windows path
213:  9b5c4ed4d8 = 213:  80156be6a0 t9200: skip tests when $PWD contains a colon
214:  c78531d0f6 = 214:  52a1291d17 mingw: kill child processes in a gentler way
216:  f534655d9c = 215:  0de7f5e28c mingw: add a Makefile target to copy test artifacts
215:  af1eece682 = 216:  181e466f8c mingw: do not call xutftowcs_path in mingw_mktemp
217:  864ec33ab4 = 217:  e03b030034 mingw: really handle SIGINT
218:  bd50ab65d3 = 218:  6dfcedf0f1 Partially un-revert "editor: save and reset terminal after calling EDITOR"
219:  9209cb6686 = 219:  98073ac070 reset: reinstate support for the deprecated --stdin option
222:  fefa706e7a = 220:  6914f366b4 Describe Git for Windows' architecture [no ci]
223:  2de8f76c0e = 221:  bdc5f96761 Modify the Code of Conduct for Git for Windows
224:  4fdce33acb = 222:  e3275ecf7b CONTRIBUTING.md: add guide for first-time contributors
225:  93b0207e5c = 223:  29a2c35bb8 README.md: Add a Windows-specific preamble
226:  8fc5573df1 = 224:  ae7a2c4dda Add an issue template
220:  5ca864827f = 225:  e890ff37e8 fsmonitor: reintroduce core.useBuiltinFSMonitor
221:  c58f0000c2 = 226:  11a01f19e4 Add a GitHub workflow to monitor component updates
227:  00ebf09456 = 227:  74d39521cf Modify the GitHub Pull Request template (to reflect Git for Windows)
228:  9e60d9f349 = 228:  4f98083fb7 .github: Add configuration for the Sentiment Bot
229:  b00c335f94 = 229:  00e9d6ef19 Document how $HOME is set on Windows
230:  4ac68b6a54 = 230:  c869f98797 SECURITY.md: document Git for Windows' policies
```

mjcheetham commented:
/git-artifacts

mjcheetham commented:
/release

dscho commented:
> Windows build fails to due the [removal of an argument from `git_check_attr`](https://github.com/git/git/commit/44451a2e5eec5360378be23e2cdbd9ecee49e14e) in `mingw.c` in .
> 
> The only call-site passed `NULL` for the removed arg; updated to drop it.
> 
> ```
> 191:  6a73477347 ! 194:  871817c12a mingw: allow to specify the symlink type in .gitattributes
>     @@ compat/mingw.c: int link(const char *oldpath, const char *newpath)
>      +	if (!check)
>      +		check = attr_check_initl("symlink", NULL);
>      +
>     -+	git_check_attr(index, NULL, link, check);
>     ++	git_check_attr(index, link, check);
>      +
>      +	value = check->items[0].value;
>      +	if (ATTR_UNSET(value))
> ```

That's the same fix as I added to `shears/seen` back on May 8th: 01eb0fbe4765c048b39924a929a1b27f95b9b32e, so I am obviously agreeing that it is the correct fix 😁
